ci/docker/Dockerfile.alpine-release

```diff
@@ -0,0 +1,51 @@
+FROM alpine:3.15 as build
+
+# creates /opensmtpd dir and makes all following commands to run in it
+# https://docs.docker.com/engine/reference/builder/#workdir
+WORKDIR /opensmtpd
+
+# install necessary packages
+RUN apk add --no-cache \
+    autoconf \
+    automake \
+    bison \
+    ca-certificates \
+    fts-dev \
+    gcc \
+    fts \
+    libevent-dev \
+    libtool \
+    libtool \
+    linux-pam-dev \
+    make \
+    musl-dev \
+    openssl \
+    openssl-dev \
+    zlib-dev
+
+# create users and directories
+# note: alpine uses busybox and useradd is not available there
+# also long flags are not available too, so sorry for the
+RUN mkdir -p /var/lib/opensmtpd/empty \
+  && adduser _smtpd -h /var/lib/opensmtpd/empty/ -D -H -s /bin/false \
+  && adduser _smtpq -h /var/lib/opensmtpd/empty/ -D -H -s /bin/false \
+  && mkdir -p /var/spool/smtpd \
+  && mkdir -p /var/mail \
+  && mkdir -p /etc/mail \
+  && chmod 711 /var/spool/smtpd
+
+# Copy contentes of the repo inside the container
+# https://docs.docker.com/engine/reference/builder/#copy
```

Comment:
I'd avoid keeping this kind of commits.  It's useless to link the documentation IMHO, if one doesn't know how a Dockerfile works they can find the doc by themselves, and we can avoid littering the file with links.

it's probably an issue with the CI dockerfiles.  I'll take care to clean them up sometime.

Reply:
Yeah this was taken from the CI file. I need to do cleanup anyway, I was thinking of splitting the && shell pipelines into separate RUN instructions to be more in line with the commits/layers philosophy. I'll wanted an initial go ahead and now I'll start on that.

Reply:
Removed unnecessary comments in the dockerfile I am adding.